apps/eew/scfinder/main.cpp

```diff
@@ -556,18 +561,42 @@ class App : public Client::StreamApplication {
 								}
 							}
 						}
-						
+						// Clone _latestMaxPGAs		
+						PGA_Data_List originBasedLatestMaxPGAs ;
+						SEISCOMP_DEBUG("Cloned PGAs:");
+						for ( size_t i = 0; i < _latestMaxPGAs.size(); ++i ) {
+							PGA_Data &pga = _latestMaxPGAs[i];
+							originBasedLatestMaxPGAs.push_back(PGA_Data(pga.get_name().c_str(),
+																		 pga.get_network().c_str(),
+																		 pga.get_channel().c_str(),
+																		 pga.get_location_code().c_str(),
+																		 Coordinate( pga.get_location().get_lat(),
+																		     		 pga.get_location().get_lon() ),
+																		 (double)pga.get_value(),
+																	 	 pga.get_timestamp()));
+
+							SEISCOMP_DEBUG("%s.%s.%s.%s %f %f %s %f",
+											pga.get_network().c_str(),
+											pga.get_name().c_str(),
+											pga.get_location_code().c_str(),
+											pga.get_channel().c_str(),
+											pga.get_location().get_lat(),
+											pga.get_location().get_lon(),
+											Core::Time(pga.get_timestamp()).iso().c_str(),
+											pga.get_value());
+						}
+
```

Comment:
@jenrandrews: is this the best way to copy the _latestMaxPGAs data ?

Reply:
Hi Fred, since PGA_Data_List objects are just vectors containing objects of PGA_Data class, I believe you can just do A = B, and A will be a deep copy of B. This may (or may not) be more efficient than your explicit copy loop, since it may not involve vector resizing. However, there is no copy constructor defined for the PGA_Data class, so something may go awry. My suggestion would be to do a quick test: try the originBasedLatestMaxPGAs = _latestMaxPGAs method, change something in one of the PGA_Data objects, and see if it shows up in the other list. If it doesn't, you're golden, otherwise keep what you have.